memote/support/basic.py

```diff
@@ -273,6 +273,36 @@ def find_duplicate_metabolites_in_compartments(model):
     return duplicates
 
 
+def find_duplicate_reactions(model):
+    """
+    Return list of reactions with duplicates.
+
+    All models should contain a unique set of reactions. This function checks
+    for and returns a list of tuples containing the duplicate reactions. An
+    example of this would be finding reactions PGI1 and PGI2, both of which
+    convert G6P(c) to F6P(c) in the same compartments.
+
+    Parameters
+    ----------
+    model : cobra.Model
+        The metabolic model under investigation
+
+    """
+    duplicates = []
+    RXN_DB = ["metanetx.reaction", "kegg.reaction", "brenda", "rhea", "biocyc",
+              "bigg.reaction"]
+    ann_rxns = [(rxn, rxn.annotation) for rxn in model.reactions
```

Comment:
This does a little duplicated work as your storing pairs of reactions and their complete annotation for every database. So if a reaction has annotations for four of those databases, for example, then you will have the same pair four times in that list.

Overall, I have two ideas to improve the below duplicate checking code:

1. Construct a list of pairs of the reaction and all its annotations in a set. I'm constructing a set of pairs of database identifier and reaction identifier here just in case two different reactions could have the same identifier in different databases. (This code is untested just off the top of my head.)

```python
ann_rxns = [(rxn, frozenset((db_id, rxn_id) for db_id in RXN_DB for rxn_id in rxn.annotation[db_id]) for rxn in model.reactions]

for (rxn_a, ann_a), (rxn_b, ann_b) in combinations(ann_rxns, 2):
    if len(ann_a & ann_b) > 0:
       duplicates.append((rxn_a, rxn_b))
```

2. The second idea is maybe for later as it involves creating RInChl keys for all the reactions. Such keys are unique based on the structure (InChl key) of the metabolites and the stoichiometry.

Reply:
I like the second option more. That way, everything is depending on the structure and stoichiometry of individual metabolites in the model.

Reply:
I totally agree but computing RInChl keys is not that easy and as far as @joaocardoso tells me the corresponding package to do so is a horrible C extension. Option 1 should be a quick improvement to the current code.

Reply:
Okay yeah, I'll try that first. My alternative was to reaction a tuple `(rxn, dict)` where dict would be a dictionary with metabolite inchikey ids, and the values would be their stoichiometry in the reaction.

Reply:
That also sounds good but we can't assume to have InChl keys for every metabolite in every model, can we?